File: TemplateAPI/TemplateAPI/src/Domain/Entities/AccountType.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TemplateAPI.Domain.Entities;

/// <summary>
///     Account Types - Third level classification
/// </summary>
public class AccountType : BaseAuditableEntity
{
    [Required] [MaxLength(50)] public string Code { get; set; } = string.Empty;

    [Required] [MaxLength(100)] public string Name { get; set; } = string.Empty;

    [MaxLength(500)] public string Description { get; set; } = string.Empty;

    public int AccountSubCategoryId { get; set; }

    [ForeignKey(nameof(AccountSubCategoryId))]
    public AccountSubCategory AccountSubCategory { get; set; } = null!;

    public NormalBalance NormalBalance { get; set; }

    public bool IsActive { get; set; } = true;

    public int DisplayOrder { get; set; }

    // Navigation
    public ICollection<Account> Accounts { get; set; } = new List<Account>();
}

File: angular-template/src/app/modules/shared/models/account-type.ts
export interface AccountTypeDTO {
  id: number;
  code: string;
  name: string;
  description: string;
  accountSubCategoryId: number;
  accountSubCategoryName: string;
  normalBalance: string;
  isActive: boolean;
  displayOrder: number;
}

export interface CreateAccountTypeCommand {
  code: string;
  name: string;
  description: string;
  accountSubCategoryId: number;
  normalBalance: NormalBalance;
  displayOrder: number;
}

export interface UpdateAccountTypeCommand {
  id: number;
  name: string;
  description: string;
  isActive: boolean;
  displayOrder: number;
}

export enum NormalBalance {
  Debit = 1,
  Credit = 2
}
